### DISCUSSION TITLE
Running Silk AndroidDemo @HEAD with latest dotnet

### DISCUSSION THREAD

**Discussion by ignatz** [dotnet/Silk.NET#1988] [Q&A]
Hi! Let me start saying that I'm rather new to dotnet so I might get some of the lingo wrong, please help me out.

First, I'd like to get a Silk App running on Android. So I went to "examples/CSharp/OpenGL Demos/AndroidDemo" as a starting point. First thing I ran into is that the TargetFramework of "net6.0-android" is no longer supported. Most of the intrnets seems to be think that newer versions should be pretty backwards compatible and thus picking a newer target framework should do the trick. When I switch to net8 (or 7) and build again, i'm getting:

```
$ dotnet run -f net8.0-android 
...
Silk.NET/src/Windowing/Silk.NET.Windowing.Sdl/obj/Debug/net7.0-android/generated/src/Org.Libsdl.App.SDLControllerManager.cs(14,43): error CS0234: The type or namespace name 'SDLHapticHandler' does not exist in the namespace 'Org.Libsdl.App' (are you missing an assembly reference?) [Silk.NET/src/Windowing/Silk.NET.Windowing.Sdl/Silk.NET.Windowing.Sdl.csproj::TargetFramework=net7.0-android]
Silk.NET/src/Windowing/Silk.NET.Windowing.Sdl/obj/Debug/net7.0-android/generated/src/Org.Libsdl.App.SDLControllerManager.cs(36,43): error CS0234: The type or namespace name 'SDLJoystickHandler' does not exist in the namespace 'Org.Libsdl.App' (are you missing an assembly reference?) [Silk.NET/src/Windowing/Silk.NET.Windowing.Sdl/Silk.NET.Windowing.Sdl.csproj::TargetFramework=net7.0-android]
...
```

Which I assume is dotnet's way of referring to undefined referrences, which seems very reasonable given the version issues. On the other hand it's surprising since it seems to be a mismatch with libsdl rather than dotnet :woman_shrugging: . Are folks able to Silk Android with dotnet8? If so, any clues?

To make things a bit more complicated, I'm running Linux (so advice would ideally go beyond "click X in Visual Studio").

**Comment by ignatz** [comment]
Eventually I overcame the namespace issues above. Not entirely sure what it was.

Then I battled the Public API Analyzer to eventually get an APK. After further battling BLUETOOTH_CONNECT permission issues and explicit broadcast receiver exports, on more recent android versions. I'm now in a place where depending on the Android SDK and architecture I'm bouncing between various mono asserts, other sigaborts, and SDL segmentation faults. I really hope, I'm just holding it wrong, otherwise is this expected?

**Reply by tweggen** [reply]
Hi, also facing some android 14 issues with silk net and dotnet 8. Could you probably give me a hint what you did to overcome the register receiver requirements of android 14? Did you replace sdls activity implementation?
